### PR TITLE
Implemented newsletter component

### PR DESCRIPTION
**Pull request by gonuguntla**

### CODE REVIEW

File: app/javascript/components/NewsLetter.jsx [hunk 1]
```diff
@@ -1,70 +1,64 @@
...
-          setTimeout(() => {
-            const newsLetterElement = document.getElementById(
-              "news-letter-msg-box"
-            );
-            if (newsLetterElement) {
-              newsLetterElement.focus();
-            }
-          }, 1);
+          setView("confirmation");
+          setConfirmationHeading("Thank you for signing up to the Newsletter!");
```
Comment (bigfishdesign13) [inline app/javascript/components/NewsLetter.jsx]: Update the message to the following.
```
Thank you for signing up for our newsletter!
```

File: app/javascript/components/NewsLetter.jsx [hunk 2]
```diff
@@ -80,60 +74,36 @@ export default function NewsLetter() {
...
+    return (
+      <NewsletterSignup
+        id="news-letter-text-input"
+        view={view}
+        isInvalidEmail={isInvalidEmail}
+        valueEmail={email}
+        onChange={handleNewsLetterEmail}
+        onSubmit={handleSubmit}
+        showPrivacyLink={false}
+        title="Sign Up for Our Newsletter"
```
Comment (bigfishdesign13) [inline app/javascript/components/NewsLetter.jsx]: Update the capitalization.
```
Sign up for our newsletter
```

File: app/javascript/stylesheets/application.scss [hunk 3]
```diff
@@ -523,3 +523,11 @@ ul.list li:before {
...
   }
 }
+
+#news-letter-success-msg {
+  #newsletter-form {
+    #submit {
+      background: var(--nypl-colors-ui-black) !important;
+    }
+  }
+}
```
Comment (bigfishdesign13) [inline app/javascript/stylesheets/application.scss]: This is working, but make sure to also include a color value for dark mode.

File: app/javascript/components/NewsLetter.jsx [hunk 4]
```diff
@@ -152,42 +122,8 @@ export default function NewsLetter() {
...
-  };
-
   return (
-    <Box
-      bg={newsLetterbg}
-      p="l"
-      ref={newsLetterMsgRef}
-      id="news-letter-success-msg"
-    >
+    <Box p="l" ref={newsLetterMsgRef} id="news-letter-success-msg">
```
Comment (bigfishdesign13) [inline app/javascript/components/NewsLetter.jsx]: Remove `p="l"`.

File: app/javascript/components/NewsLetter.jsx [hunk 5]
```diff
@@ -80,60 +74,36 @@ export default function NewsLetter() {
...
+            <Link
+              type="action"
+              target="_blank"
+              href="mailto:help@mylibrarynyc.org"
+            >
+              contact our e-mail team.
+            </Link>
+          </Text>
+        }
+        confirmationText="Check your email to learn about teacher sets, best practices & exclusive events."
```
Comment (bigfishdesign13) [inline app/javascript/components/NewsLetter.jsx]: Add a comma after `best practices` and change the ampersand (&) to `and`

File: app/javascript/components/NewsLetter.jsx [hunk 6]
```diff
@@ -80,60 +74,36 @@ export default function NewsLetter() {
...
+      <NewsletterSignup
+        id="news-letter-text-input"
+        view={view}
+        isInvalidEmail={isInvalidEmail}
+        valueEmail={email}
+        onChange={handleNewsLetterEmail}
+        onSubmit={handleSubmit}
+        showPrivacyLink={false}
+        title="Sign Up for Our Newsletter"
+        descriptionText="Learn about new teacher sets, best practices & exclusive events when you sign up for the MyLibraryNYC Newsletter!"
```
Comment (bigfishdesign13) [inline app/javascript/components/NewsLetter.jsx]: Add a comma after `best practices` and change the ampersand (&) to `and`.  Also, use a lowercase `n` for `newsletter`.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Looks good.  I just have some changes for the text and capitalization.  Also, it seems like the signup API is relatively slow.  Is it always like that?

File: app/javascript/components/NewsLetter.jsx (hunk 1 above)
Comment (gonuguntla) [reply app/javascript/components/NewsLetter.jsx]: @bigfishdesign13  Yes Marty New letter signup is slow. We are checking from google sheets.

**Review by bigfishdesign13 — APPROVED** [review]
The code updates look good.  Please let me know once the updates have been pushed to the QA site, so I can complete the VQA.

**Pull request merged**